frontend/src/components/custom-tools/pdf-viewer/PdfViewer.jsx

```diff
@@ -28,6 +34,19 @@ function PdfViewer({ fileUrl, highlightData }) {
     });
   }
 
+  // Jump to page when highlightData changes
+  useEffect(() => {
+    if (highlightData && highlightData.length > 0) {
+      highlightData = removeZerosAndDeleteIfAllZero(highlightData);
+      const pageNumber = highlightData[0][0]; // Assume highlightData[0][0] is the page number
+      if (pageNumber !== null && jumpToPage) {
+        setTimeout(() => {
+          jumpToPage(pageNumber); // jumpToPage is 0-indexed, so subtract 1
+        }, 100); // Add a slight delay to ensure proper page rendering
```

Comment:
@vishnuszipstack Could you please explain why the `setTimeout` is necessary? From what I understand, the library’s `jumpToPage` function should handle this effectively on its own.

Reply:
@tahierhussain jumpToPage is not handled and it was not working fine when delay is not there. for example jumpto page 4 on highlight and if user tries to jump back to page 1 on highlight not happening if delay is not there.